src/Coordination/KeeperDispatcher.cpp

```diff
@@ -445,11 +452,17 @@ void KeeperDispatcher::shutdown()
     try
     {
         {
-            if (shutdown_called.exchange(true))
+            if (keeper_context->shutdown_called.exchange(true))
                 return;
 
             LOG_DEBUG(log, "Shutting down storage dispatcher");
 
+            our_last_committed_log_idx = std::numeric_limits<uint64_t>::max();
+            our_last_committed_log_idx.notify_all();
+
+            keeper_context->local_logs_preprocessed = true;
+            keeper_context->local_logs_preprocessed.notify_all();
```

Comment:
I'm slightly afraid of atomic synchronization, but do we wait somewhere for this flag value?

Reply:
needs to be removed, leftover from previous implementation